### PR TITLE
Adding deprecation warning to explain_prediction.

### LINKED ISSUE #1822: Deprecate the explain_prediction function
Once #1818 is merged, the following two function calls will be the same:

```python
explain_prediction(pipeline, input_features, y, index_to_explain=3)
```

```python
explain_predictions(pipeline, input_features, y, indices_to_explain=[3])
```

I think this makes `explain_prediciton` redundant.

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #1822

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -25,6 +25,11 @@ Release Notes
         * Pin graphviz version for windows builds :pr:`1847`
         * Unpin graphviz version for windows builds :pr:`1851`
 
+.. warning::
+
+    **Breaking Changes**
```
Comment (angela97lin) [inline docs/source/release_notes.rst:30]: This is only listed as a breaking change but not in the other sections of the release notes? :O Is this intentional 👀

**Review by angela97lin — APPROVED** [review]
LGTM! :shipit: 

Do we have an issue filed for removing `explain_prediction` yet? :O

File: docs/source/release_notes.rst (hunk 1 above)
Comment (freddyaboulton) [reply docs/source/release_notes.rst:30]: I originally had it in both but then felt it was redundant. Do we have a guideline in place for this?

**Comment by freddyaboulton** [comment]
@angela97lin Yes! We have #1832

**Review by bchen1116 — APPROVED** [review]
Nice!

**Comment by angela97lin** [comment]
@freddyaboulton Amazing, didn't realize it was filed a while ago. Thank you!

**Review by jeremyliweishih — APPROVED** [review]
🚢

**Comment by dsherry** [comment]
Passing RTD build [here](https://readthedocs.com/projects/feature-labs-inc-evalml/builds/567152/).

File: docs/source/release_notes.rst (hunk 1 above)
Comment (dsherry) [reply docs/source/release_notes.rst:32]: To date, we've listed in both places. And a single PR could contain multiple lines of change in release notes. However for something this small, I think this is fine.

**Review by dsherry — APPROVED** [review]
Looks good!

**Review by chukarsten — COMMENTED** [review]
I really love the simple PRs.

**Comment by chukarsten** [comment]
> LGTM! :shipit:
> 
> Do we have an issue filed for removing `explain_prediction` yet? :O
We sure do!  It's targeted for next sprint.

https://github.com/alteryx/evalml/issues/1832

**Pull request merged**